simulator/docker/update-env.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# This script creates a '.env' file that is used for docker-compose as input for environment variables
+# for the simulator services.
+
+echo "Creating .env file for simulator services..."
+
+# Generate .env file with default values
+cat > .env << EOL
+GRPC_SERVER_ADDRESS=block-node-server
+PROMETHEUS_ENDPOINT_ENABLED=true
+
+# For publisher service
+PUBLISHER_BLOCK_STREAM_SIMULATOR_MODE=PUBLISHER
+PUBLISHER_PROMETHEUS_ENDPOINT_PORT_NUMBER=9998
+
+# For consumer service
+CONSUMER_BLOCK_STREAM_SIMULATOR_MODE=CONSUMER
+CONSUMER_PROMETHEUS_ENDPOINT_PORT_NUMBER=9997
+EOL
+
```

Comment:
here we could add a conditional:

```shell
if [ true = "$is_debug" ]; then
  # The server will wait for the debugger to attach on port 5005
  # JProfiler can attach on port 8849
  echo "SERVER_OPTS='-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005' -agentpath:/path/to/libjprofilerti.so=port=8849 " >> .env
fi
```

In this way, if you update the env conditionally directly from gradle task (like in the server) we can very easily enable debugging. And you can simply create the `startDockerContainerDebug` task also in the simulator so we can bind to it remotely. I would not reuse the `5005` port though, the agentlib path to jprofiler could be dropped as well I think.

Reply:
Answered here: https://github.com/hashgraph/hedera-block-node/pull/416#discussion_r1893164374